### ISSUE TITLE
Not working on the visa dataset

### ISSUE THREAD

**Issue by ardaxz99** [M-3LAB/open-iad#10]
Your code is not working for the visa dataset. Visa is structure as

|----VisA
|-----|-- split_csv
|-----|-----|--- 1cls.csv
|-----|-----|--- ...
|-----|-- candle
|-----|-----|--- Data
|-----|-----|-----|----- Images
|-----|-----|-----|--------|------ Anomaly 
|-----|-----|-----|--------|------ Normal 
|-----|-----|-----|----- Masks
|-----|-----|-----|--------|------ Anomaly 
|-----|-----|--- image_anno.csv
|-----|-- capsules
|-----|-----|----- ...

But your code to load the visa dataset uses the mvtec format:

you need to change:

    def load_dataset(self):
        # input x, label y, [0, 1], good is 0 and bad is 1, mask is ground truth
        # train directory: only good cases
        # test directory: bad and good cases 
        # ground truth directory: only bad case

        # get classes in each task group
        # If num_task is 15, each task constain each class
        self.class_in_task = self.split_chunks(self.class_name, self.num_task)
        # get data
        for id, class_in_task in enumerate(self.class_in_task):
            x, y, mask = [], [], []
            for class_name in class_in_task:
                img_dir = os.path.join(self.data_path, class_name, self.phase)
                gt_dir = os.path.join(self.data_path, class_name, 'ground_truth') 

                img_types = sorted(os.listdir(img_dir))
                for img_type in img_types:

                    # load images
                    img_type_dir = os.path.join(img_dir, img_type)
                    if not os.path.isdir(img_type_dir):
                        continue
                    img_path_list = sorted([os.path.join(img_type_dir, f)
                                            for f in os.listdir(img_type_dir)
                                            if f.endswith('.JPG')])
                    x.extend(img_path_list)

                    if img_type == 'good':
                        y.extend([0] * len(img_path_list))
                        mask.extend([None] * len(img_path_list))
                    else:
                        y.extend([1] * len(img_path_list))
                        gt_type_dir = os.path.join(gt_dir, img_type)
                        img_name_list = [os.path.splitext(os.path.basename(f))[0] for f in img_path_list]
                        gt_path_list = [os.path.join(gt_type_dir, img_fname + '.png')
                                        for img_fname in img_name_list]
                        mask.extend(gt_path_list)
            
            task_id = [id for i in range(len(x))]
            self.sample_num_in_task.append(len(x))

            self.imgs_list.extend(x)
            self.labels_list.extend(y)
            self.masks_list.extend(mask)
            self.task_ids_list.extend(task_id) 

Pls fix your code. I am surprised how could you publish your code without testing.

**Comment by jiaq-liu** [comment]
@ardaxz99 hello, maybe you noticed the visa dataset has no training dataset or test dataset. You should run `python ./utils/prepare_data.py --split-type 1cls --data-folder ./VisA --save-folder ./VisA_pytorch --split-file ./split_csv/1cls.csv` according to instruction in [VisA](https://github.com/amazon-science/spot-diff?tab=readme-ov-file#data-preparation) to prepare the dataset.
